Assets/HoloToolkit/Input/Scripts/GestureManager.cs

```diff
@@ -258,8 +264,17 @@ void LateUpdate()
 
             bool focusedChanged = FocusedObject != newFocusedObject;
 
+            if (focusedChanged)
+            {
+                // If the currently focused object doesn't match the new focused object, cancel the current gesture.
+                // Start looking for new gestures.  This is to prevent applying gestures from one hologram to another.
+                gestureRecognizer.CancelGestures();
+                FocusedObject = newFocusedObject;
+                gestureRecognizer.StartCapturingGestures();
+            }
+
 #if UNITY_EDITOR
-            if (Input.GetMouseButtonDown(1) || Input.GetKeyDown(EditorSelectKey))
+            if ((Input.GetMouseButtonDown(1) || Input.GetKeyDown(EditorSelectKey)) && !focusedChanged)
```

Comment:
This is mainly for Input from the editor or companion app I think. So OnRecognitionEnded and OnRecognitionStarted started has nothing to do with input from the HoloLens.  We would only want to process the Tap if we've got a new focused object.